docs/run-node/interact-node.md

```diff
@@ -44,3 +47,109 @@ simd query staking delegations-to $(simd keys show my_validator --bech val -a --
+
+Since the code generation library largely depends on your own tech stack, we will only present two alternatives:
+
+- `grpcurl` for generic debugging and testing,
+- CosmJS for JavaScript/TypeScript developers.
+
+### grpcurl: Reflection, Queries, and Simulation
+
+[grpcurl])https://github.com/fullstorydev/grpcurl is like `curl` but for gRPC. It is also available as a Go library, but we will use it only as a CLI command for debugging and testing purposes. Follow the instructions in the previous link to install it.
+
+Assuming you have a local node running (either a localnet, or connected a live network), you should be able to run the following command to list the Protobuf services available (you can replace `localhost:9000` by the gRPC server endpoint of another node, which is configured under the `grpc.address` field inside `app.toml`):
+
+```bash
+grpcurl -plaintext localhost:9090 list
+```
+
+You should see a list of gRPC services, like `cosmos.bank.v1beta1.Query`. This is called reflection, which is a Protobuf endpoint returning a description of all available endpoints. Each of these represents a different Protobuf service, and each service exposes multiple RPC methods you can query against.
+
+In the Cosmos SDK, we use [gogoprotobuf](https://github.com/gogo/protobuf) for code generation, and [grpc-go](https://github.com/grpc/grpc-go) for creating the gRPC server. Unfortunately, these two don't play well together, and more in-depth reflection (such as using grpcurl's `describe`) is not possible. See [this issue](https://github.com/grpc/grpc-go/issues/1873) for more info.
+
+Instead, we need to manually pass the reference to relevant `.proto` files. For example:
+
+```bash
+grpcurl \
+    -import-path ./proto \                              # Import these proto files too
+    -import-path ./third_party/proto \                  # Import these proto files too
+    -proto ./proto/cosmos/bank/v1beta1/query.proto \    # That's the proto file with the description of your service
+    localhost:9090 \
+    describe cosmos.bank.v1beta1.Query                  # Service we want to inspect
+```
+
+Once the Protobuf definitions are given, making a gRPC query is then straightforward, by calling the correct `Query` service RPC method, and by passing the request argument as data (`-d` flag):
+
+```bash
+grpcurl \
+    -plaintext
+    -import-path ./proto \
+    -import-path ./third_party/proto \
+    -proto ./proto/cosmos/bank/v1beta1/query.proto \
+    -d '{"address":"$MY_VALIDATOR"}' \
+    localhost:9090 \
+    cosmos.bank.v1beta1.Query/AllBalances
+```
+
+The list of all available gRPC query endpoints is [coming soon](https://github.com/cosmos/cosmos-sdk/issues/7786).
```

Comment:
it would be good to track that in #7786 so we don't forget to update this doc once it's ready.